[EventSource] Stop allowing trailing junk when parsing retry delay

cdumez opened the pull request:
#### 8d4846adfa1f441dac946b055e18ed5c74ba2cf9
<pre>
[EventSource] Stop allowing trailing junk when parsing retry delay
<a href="https://bugs.webkit.org/show_bug.cgi?id=243227">https://bugs.webkit.org/show_bug.cgi?id=243227</a>

Reviewed by Darin Adler.

Stop allowing trailing junk when parsing retry delay for EventSource:
- <a href="https://html.spec.whatwg.org/multipage/server-sent-events.html#processField">https://html.spec.whatwg.org/multipage/server-sent-events.html#processField</a>

This aligns our behavior with Blink and Gecko.

* LayoutTests/imported/w3c/web-platform-tests/eventsource/format-field-retry-bogus.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/eventsource/format-field-retry-bogus.any.worker-expected.txt:
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):

Canonical link: <a href="https://commits.webkit.org/252885@main">https://commits.webkit.org/252885@main</a>
</pre>

darinadler commented on Source/WebCore/page/EventSource.cpp:
```diff
@@ -382,9 +382,7 @@ void EventSource::parseEventStreamLine(unsigned position, std::optional<unsigned
             m_reconnectDelay = defaultReconnectDelay;
         else {
-            // FIXME: Do we really want to ignore trailing junk here?
-            // FIXME: When we can't parse the value, should we really leave m_reconnectDelay alone? Shouldn't we set it to defaultReconnectDelay?
```
Is there a WPT test covering the answer to this question?

cdumez replied:
No but our behavior matches other browsers. I'll add such test and upstream it.

cdumez replied:
https://github.com/web-platform-tests/wpt/pull/35249

cdumez committed:
[EventSource] Stop allowing trailing junk when parsing retry delay
https://bugs.webkit.org/show_bug.cgi?id=243227

Reviewed by Darin Adler.

Stop allowing trailing junk when parsing retry delay for EventSource:
- https://html.spec.whatwg.org/multipage/server-sent-events.html#processField

This aligns our behavior with Blink and Gecko.

* LayoutTests/imported/w3c/web-platform-tests/eventsource/format-field-retry-bogus.any-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/eventsource/format-field-retry-bogus.any.worker-expected.txt:
* Source/WebCore/page/EventSource.cpp:
(WebCore::EventSource::parseEventStreamLine):

Canonical link: https://commits.webkit.org/252885@main
